decred/dcrd
# multi: Enable vote for DCP0004.

**This requires PR #1578.**

This implements the agenda for voting on the sequence lock fixes as defined in DCP0004 (not completed yet -- will be linked here when complete) along with consensus tests and `mempool` acceptance tests to ensure its correctness.

It also modifies the `mempool` to conditionally treat all transactions with enabled sequence locks as non standard until the vote passes at which point the will become standard with the modified semantics enforced.

The following is an overview of the changes:

- Generate new version blocks and reject old version blocks after a super majority has been reached
  - New block version on `mainnet` is version 6
  - New block version on `testnet` is version 7
- Introduce a convenience function for determining if the vote passed and is now active
- Enforce modified sequence lock semantics in accordance with the state of the vote
- Modify the more strict standardness checks (acceptance to the `mempool` and relay) to enforce DCP0004 in accordance with the state of the vote
  - Make all transactions with enabled sequence locks non standard until the agenda vote passes
  - Add tests to ensure acceptance and relay behave according to the aforementioned description
- Add tests for determining if the agenda is active for both `mainnet` and `testnet`
- Add tests to ensure the corrected sequence locks are handled properly depending on the result of the vote
